src/core/probe/bpf/include/common_defs.h

```diff
@@ -32,6 +31,52 @@ static __always_inline bool collection_enabled() {
 #define COMMON_SECTION_CORE	0
 #define COMMON_SECTION_TASK	1
 
+/* Current log level. */
+const volatile u8 log_level = 0;
+
+/* Aligned with the log crate. */
+enum {
+	LOG_ERROR = 1,
+	LOG_WARN,
+	LOG_INFO,
+	LOG_DEBUG,
+	LOG_TRACE,
+};
+
+/* Checks if the level will be reported. */
+#define retis_log_enabled(__lvl)			\
+({							\
+	bool active = false;				\
+	if (log_level && __lvl <= log_level)		\
+		active = true;				\
+	active;						\
+})
+
+/* Log macros must be used carefully and preferrably in the
+ * {error,slow} path.
+ * Useful exceptions must use a high log level (ideally LOG_TRACE).
+ */
+#define retis_log(lvl, fmt, args...)					\
+({									\
+	struct retis_log_event *__log = NULL;				\
+	if (retis_log_enabled(lvl)) {					\
+		__log = bpf_ringbuf_reserve(&log_map, sizeof(struct retis_log_event), 0); \
+		if (__log) {						\
```

Comment:
If `__log` is `NULL`, could we use a counter to inform the user at the end of the collect? And we also probably don't need to return `__log` at the end of `retis_log`.

Reply:
the problem is that counters need pid (for usdt probes, 0 otherwise) and `sym_addr` (target).
Probably we should wrap logs functions and, based on the "return" increase the counter.
As of now, message drop should never happen (or if it happens we'll see a good number of repeating logs before).
I think this will have some follow-ups anyways, as we might want to consider some rate limit mechanism and at this point global counters (or different approaches). WDYT?

Reply:
Makes sense, let's not use a counter for now.